daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2269,7 +2333,7 @@ func NewDeviceSet(root string, doInit bool, options []string, uidMaps, gidMaps [
 		metaDataLoopbackSize:  defaultMetaDataLoopbackSize,
 		baseFsSize:            defaultBaseFsSize,
 		overrideUdevSyncCheck: defaultUdevSyncOverride,
-		filesystem:            "ext4",
+		defaultFilesystem:     determineDefaultFS(),
```

Comment:
whatever works. The way I'll test it is by upgrading docker with this patch and see if my images are still there and runnable.